examples/load-gen.ts

```diff
@@ -133,69 +142,93 @@ class BasicJavaScriptLoadGen {
     client: SimpleCacheClient,
     loadGenContext: BasicJavasScriptLoadGenContext,
     workerId: number,
-    numOperations: number
+    totalMsToRun: number,
+    delayMillisBetweenRequests: number
   ): Promise<void> {
-    for (let i = 1; i <= numOperations; i++) {
-      await this.issueAsyncSetGet(client, loadGenContext, workerId, i);
+    let finished = false;
+    const finish = () => (finished = true);
+    setTimeout(finish, totalMsToRun);
+    const intervalId = setInterval(() => {
+      this.logStats(loadGenContext);
+    }, this.printStatsEveryNMs);
+
+    let i = 1;
+    for (;;) {
+      await this.issueAsyncSetGet(
+        client,
+        loadGenContext,
+        workerId,
+        i,
+        delayMillisBetweenRequests
+      );
+
+      if (finished) {
+        clearInterval(intervalId);
+        this.logStats(loadGenContext);
+        return;
+      }
 
-      if (loadGenContext.globalRequestCount % 1000 === 0) {
-        this.logger.info(`
+      i++;
+    }
+  }
+
+  private logStats(loadGenContext: BasicJavasScriptLoadGenContext): void {
+    this.logger.info(`
 cumulative stats:
-   total requests: ${
-     loadGenContext.globalRequestCount
-   } (${BasicJavaScriptLoadGen.tps(
-          loadGenContext,
-          loadGenContext.globalRequestCount
-        )} tps)
-           success: ${
-             loadGenContext.globalSuccessCount
-           } (${BasicJavaScriptLoadGen.percentRequests(
-          loadGenContext,
-          loadGenContext.globalSuccessCount
-        )}%) (${BasicJavaScriptLoadGen.tps(
-          loadGenContext,
-          loadGenContext.globalSuccessCount
-        )} tps)
-       unavailable: ${
-         loadGenContext.globalUnavailableCount
+total requests: ${
```

Comment:
not a huge deal but the indentation here was intentional, so that we were nesting several subsections underneath 'cumulative stats' header in the output

Reply:
I think that's prettier's fault.

Reply:
It doesn't seem that indentation was consistently maintained anyway.